pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
```

Comment:
I actually think that HeldBy should be a field in AcquireInfo (or an annotation) until we have a consistent way of handling object ownership.

From the Chubby [paper](http://static.googleusercontent.com/media/research.google.com/en//archive/chubby-osdi06.pdf):

> both GFS andBigtable use Chubby as a well-known and available location to store a small amount of meta-data; in effect they use Chubby as the root of their distributed data structures.

.e.g. kafka stores last processed offsets in parition leases.

@bgrant0607 will have opinions on this